### PR TITLE
Add snippets for DataFrames

### PR DESCRIPTION
**Pull request by Jualns**
add somethings functions of DataFrames

### CODE REVIEW

File: data.json [hunk 1]
```diff
@@ -11,6 +11,27 @@
       "tabular-data"
     ]
   },
+  {
+    "command": "propertynames(df)",
+    "description": "Column names as Symbols use the propertynames function",
```
Comment (abelsiqueira) [inline data.json:16]:
```suggestion
    "description": "Get the column names as Symbols",
```

File: data.json [hunk 2]
```diff
@@ -11,6 +11,27 @@
...
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "Columns symbol names",
```
Comment (abelsiqueira) [inline data.json:20]:
```suggestion
      "column-names",
```

File: data.json [hunk 3]
```diff
@@ -11,6 +11,27 @@
...
+      "symbol names"
```
Comment (abelsiqueira) [inline data.json:21]:
```suggestion
      "columns"
```

File: data.json [hunk 4]
```diff
@@ -11,6 +11,27 @@
...
+    ]
+  },
+  {
+    "command": "names(df)",
+    "description": "Column names can be obtained as string using the names function",
```
Comment (abelsiqueira) [inline data.json:26]: There's no need to mention the function again. Only what the snippet does.

File: data.json [hunk 5]
```diff
@@ -11,6 +11,27 @@
...
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "Columns names",
+      "String names columns",
+      "names"
```
Comment (abelsiqueira) [inline data.json:32]: So far I'm not sure how I'm gonna use the tags, but it's probably better if we use them lowercase and separated by `-`.

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks a lot. I requested a few changes that apply to the whole PR, so I will finish checking everything afterwards. Another recommendation is to use imperative phrases, such as "Get the column names".

File: data.json [hunk 6]
```diff
@@ -11,6 +11,27 @@
...
+      "columns"
+    ]
+  },
+  {
+    "command": "names(df)",
+    "description": "Column names can be obtained as string using the names function",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "Columns names",
```
Comment (Jualns) [inline data.json:30]:
```suggestion
      "columns-names",
```

File: data.json [hunk 7]
```diff
@@ -11,6 +11,27 @@
...
+      "String names columns",
```
Comment (Jualns) [inline data.json:31]:
```suggestion
      "name-columns",
```

**Review by Jualns — COMMENTED** [review]
I adjusted your recommendations.

File: data.json [hunk 8]
```diff
@@ -22,6 +43,64 @@
       "data"
     ]
   },
+  {
+    "command": "last(df, X)",
+    "description": "The last functions can be used to look at the last X rows of a data frame",
```
Comment (abelsiqueira) [inline data.json:48]:
```suggestion
    "description": "Look at the last X rows of a data frame",
```

File: data.json [hunk 9]
```diff
@@ -22,6 +43,64 @@
...
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "last",
+      "last rows",
+      "print rows",
+      "rows"
```
Comment (abelsiqueira) [inline data.json:55]:
```suggestion
      "last-rows",
      "print-rows",
      "rows"
```

File: data.json [hunk 10]
```diff
@@ -11,6 +11,27 @@
...
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "column-names",
+      "columns"
+    ]
+  },
+  {
+    "command": "names(df)",
+    "description": "Column names can be obtained as string using the names function",
```
Comment (Jualns) [inline data.json:26]:
```suggestion
    "description": "Obtain columns name",
```

File: data.json [hunk 11]
```diff
@@ -22,6 +43,64 @@
...
+  },
+  {
+    "command": "first(df, X)",
+    "description": "The first functions can be used to look at the first X rows of a data frame",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "first",
+      "first rows",
+      "print rows",
```
Comment (Jualns) [inline data.json:66]:
```suggestion
    "description": "Look at the firs X rows of a data frame",
    "kind": "snippet",
    "package": "DataFrames",
    "tags": [
      "first",
      "first-rows",
      "print-rows",
```

File: data.json [hunk 12]
```diff
@@ -22,6 +43,64 @@
...
+      "rows"
+    ]
+  },
+  {
+    "command": "df.A = 1:8",
+    "description": "It is also possible to start with an empty DataFrame and add columns to it one by one",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "add columns",
```
Comment (Jualns) [inline data.json:76]:
```suggestion
    "description": "Add columns to it one by one",
    "kind": "snippet",
    "package": "DataFrames",
    "tags": [
      "add-columns",
```

File: data.json [hunk 13]
```diff
@@ -22,6 +43,64 @@
...
+      "empty"
+    ]
+  },
+  {
+    "command": "push!(df, (1,\"M\",...))",
+    "description": "Rows can then be added as tuples or vectors, where the order of elements matches that of columns",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "add rows",
```
Comment (Jualns) [inline data.json:87]:
```suggestion
    "description": "Add Rows as tuples or vectors, where the order of elements matches that of columns",
    "kind": "snippet",
    "package": "DataFrames",
    "tags": [
      "add-rows",
```

File: data.json [hunk 14]
```diff
@@ -22,6 +43,64 @@
...
+    "command": "show(df, allrows=true) and show(df,allcols=true)",
+    "description": "Printing options can be adjusted by calling the show function manually: if allrows=true prints all rows even if they do not fit on screen and if allcols=true does the same for columns.",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "print dataframe",
+      "show",
+      "show all columns",
+      "show all rows",
+      "show dataframe"
```
Comment (Jualns) [inline data.json:101]:
```suggestion
    "description": "Printing all rows or columns, just use allrows=true or allcols=true.",
    "kind": "snippet",
    "package": "DataFrames",
    "tags": [
      "print-dataframe",
      "show",
      "show-all-columns",
      "show-all-rows",
      "show-dataframe"
```

File: data.json [hunk 15]
```diff
@@ -11,6 +11,27 @@
...
+  },
+  {
+    "command": "names(df)",
+    "description": "Get the column names",
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "columns-name",
+      "columns-name",
+      "names"
```
Comment (abelsiqueira) [inline data.json:32]:
```suggestion
      "column-names",
      "names"
```

File: data.json [hunk 16]
```diff
@@ -22,6 +43,64 @@
...
+    "tags": [
+      "first",
+      "first-rows",
+      "print-rows",
+      "rows"
+    ]
+  },
+  {
+    "command": "df.A = 1:8",
+    "description": "Add columns to it one by one in DataFrame",
```
Comment (abelsiqueira) [inline data.json:72]:
```suggestion
    "description": "Add a column named `A` to the the DataFrame `df`",
```

File: data.json [hunk 17]
```diff
@@ -22,6 +43,64 @@
...
+    "package": "DataFrames",
+    "tags": [
+      "add-columns",
+      "columns",
+      "empty"
+    ]
+  },
+  {
+    "command": "push!(df, (1,\"M\",...))",
+    "description": "Add Rows as tuples or vectors, where the order of elements matches that of columns",
```
Comment (abelsiqueira) [inline data.json:83]:
```suggestion
    "description": "Add a new row as a tuple or vector, where the order of elements matches that of the columns of `df`",
```

File: data.json [hunk 18]
```diff
@@ -22,6 +43,64 @@
...
+    "kind": "snippet",
+    "package": "DataFrames",
+    "tags": [
+      "print-dataframe",
+      "show",
+      "show-all-columns",
+      "show-all-rows",
+      "show-dataframe"
+    ]
+  },
```
Comment (abelsiqueira) [inline data.json:103]:
```suggestion
  {
    "command": "show(df, allrows=true, allcols=true)",
    "description": "Print all rows and/or columns of the DataFrame `df`",
    "kind": "snippet",
    "package": "DataFrames",
    "tags": [
      "print-dataframe",
      "show",
      "show-all-columns",
      "show-all-rows",
      "show-dataframe"
    ]
  },
```

**Comment by abelsiqueira** [comment]
@all-contributors please add @Jualns for data

**Pull request merged**